editor/selectors.js

```diff
@@ -170,14 +373,39 @@ export function isBlockSelected( state, uid ) {
 	return state.selectedBlock.uid === uid;
 }
 
+/**
+ * Returns true if the unique ID occurs within the block multi-selection, or
+ * false otherwise.
+ *
+ * @param  {Object} state Global application state
+ * @param  {Object} uid   Block unique ID
+ * @return {Boolean}      Whether block is in multi-selection set
+ */
 export function isBlockMultiSelected( state, uid ) {
 	return getSelectedBlocks( state ).indexOf( uid ) !== -1;
 }
 
+/**
+ * Returns true if the cursor is hovering the block corresponding to the
+ * specified unique ID, or false otherwise.
+ *
+ * @param  {Object} state Global application state
+ * @param  {Object} uid   Block unique ID
+ * @return {Boolean}      Whether block is hovered
+ */
 export function isBlockHovered( state, uid ) {
 	return state.hoveredBlock === uid;
 }
 
+/**
+ * Returns focus state of the block corresponding to the specified unique ID,
+ * or null if the block is not selected. Focus state defaults to an empty
+ * object.
```

Comment:
>What other object shapes might be returned by the focus state?

As far as I recall the original intention, it's up to the block to manage the focus object how they see fit. For example differentiating between Editables of the same block:

https://github.com/WordPress/gutenberg/blob/1e46489/blocks/library/quote/index.js#L115

This is one thing I'd like to see refactored, ideally with less block implementer intervention. Relates to #878

cc @youknowriad